### ISSUE TITLE
problem with greatest factorial factorization

### ISSUE THREAD

**Issue by akritas** [sympy/sympy#8926]
Consider the polynomial 
```
p = x**18 + 11*x**17 + 30_x__16 - 38_x**15 - 148*x**14 + 316_x__13 + 378_x**12 - 1250*x**11 + 415_x__10 + 1365_x**9 - 1736*x**8 + 584_x__7 + 500_x**6 - 1692*x**5 + 1328_x__4 + 704_x**3 - 768*x**2
```
with
```
gff_list(p) = [(x**6 + 6*x**5 + x**4 - 22*x**3 + 30_x__2 + 16_x - 32, 1), (x**3 + 4*x**2 + x + 4, 2), (x*_2 + 2_x + 1, 3)]
```
However, 
```
p != ff(x**6 + 6*x**5 + x**4 - 22*x**3 + 30_x__2 + 16_x - 32, 1) \* ff(x**3 + 4*x**2 + x + 4, 2) \* ff(x*_2 + 2_x + 1, 3)
```
WHY??

**Comment by postvakje** [comment]
The problem is that currently ff(p(x),3) returns p(x)*(p(x)-1)*(p(x)-1) whereas for the factors in gff_list requires p(x)*p(x-1)*p(x-2).  These two expressions are the same if p is a polynomial of order 1, but in your case the factors are of higher order.  I submitted a PR #8941 to address this issue.

**Comment by akritas** [comment]
great!

**Comment by oscarbenjamin** [comment]
I think this was supposed to be closed by #8941
